### ISSUE TITLE
flutter doctor and flutter run don't recognize cocoapods installation

### ISSUE THREAD

**Issue by Stephanie-Limber** [flutter/flutter#149638]
### Steps to reproduce

This may be related to a common issue and I tried all of the solutions I could find, but I apologize if I missed one.

**System setup:**

- Macbook Pro, M1 Max chip
- Terminal is set to “Open using Rosetta”
- VS Code is set to “Open using Rosetta”

Versions--
MacOS = Sonoma 14.5
flutter = 3.7.8
homebrew = 4.3.3
cocoapods = 1.15.2
XCode = 15.4
VS Code = 1.89.1
Terminal = 2.14
ruby = ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.x86_64-darwin23]
gem = 3.0.3.1
git = 2.37.2

```
➜ ~ git:(main) ✗ which pod
/usr/local/bin/pod
```

**What I've tried:**

A) Flutter with Apple Silicon (https://github.com/flutter/flutter/blob/master/docs/platforms/desktop/macos/Developing-with-Flutter-on-Apple-Silicon.md) - Flutter 2.5 or later (running Flutter 3.7.8) and using Rosetta 2 (Terminal set to “Open using Rosetta”).
```
sudo gem install cocoapods
sudo gem uninstall ffi && sudo gem install ffi -- --enable-libffi-alloc
```

B) Uninstall Ruby and Cocoapods, then reinstall Cocoapods and ffi
Uninstall Ruby:
`brew uninstall ruby --force `
Uninstall CocoaPods.
`gem list --local | grep cocoapods. `
Then uninstall them one by one, in my case: sudo gem uninstall cocoapods
```
sudo gem uninstall cocoapods-core
sudo gem uninstall cocoapods-try
sudo gem uninstall cocoapods-trunk
sudo gem uninstall cocoapods-search
sudo gem uninstall cocoapods-plugins
sudo gem uninstall cocoapods-downloader
sudo gem uninstall cocoapods-deintegrate
```
Restart machine
Reinstall ffi and then CocoaPods:
```
sudo gem uninstall ffi && sudo gem install ffi -- --enable-libffi-alloc
sudo gem install cocoapods
```

C) Use homebrew to install cocoapods
```
sudo gem uninstall cocoapods
sudo gem uninstall cocoapods-core
sudo gem uninstall cocoapods-try
sudo gem uninstall cocoapods-trunk
sudo gem uninstall cocoapods-search
sudo gem uninstall cocoapods-plugins
sudo gem uninstall cocoapods-downloader
sudo gem uninstall cocoapods-deintegrate

brew install cocoapods
```

D) Make sure Command Line Tools are installed
```
➜ ~ git:(main) ✗ xcode-select --install
xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates
```

No change after any solution attempted.

### Actual results

Interestingly, flutter pub get and pod install both seem to work as expected.  I can also build the app in XCode after running those two commands.  
```
➜  app git:(app-branch) flutter pub get
Running "flutter pub get" in app...
Resolving dependencies... 
  assets_audio_player 3.0.6 (3.1.1 available)
  ...[dependencies removed to be concise]
  xdg_directories 0.2.0+3 (1.0.4 available)
Got dependencies!
➜  app git:(app-branch) cd ios
➜  ios git:(app-branch) pod install
Analyzing dependencies
Downloading dependencies
Generating Pods project
Integrating client project
Pod installation complete! There are 8 dependencies from the Podfile and 13 total pods installed.
```

But flutter doctor -v and flutter run -v return errors about cocoapods not being installed.  'flutter doctor' also says the flutter and dart binaries are missing from my path.  But echo $PATH shows they are in my PATH:

```
➜ ~ git:(main) ✗ echo $PATH
/usr/local/opt/ruby/bin:/Users/$USER/.gem/bin:/Users/$USER/bin:/usr/local/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/opt/homebrew/opt/mongodb-community@5.0/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/Library/TeX/texbin:/Users/$USER/Desktop/flutter/bin:/Users/$USER/Desktop/flutter/bin/cache/dart-sdk/bin
```

End of flutter run -v output:

```
[  +15 ms] executing: which pod
[   +8 ms] Warning: CocoaPods not installed. Skipping pod install.
             CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart
side.
             Without CocoaPods, plugins will not work on iOS or macOS.
             For more info, see https://flutter.dev/platform-plugins
           To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.
           
[   +2 ms] CocoaPods not installed or not in valid state.
[        ] Error launching application on iPhone 15.
[+1512 ms] "flutter run" took 10,279ms.
[   +1 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:722:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1257:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command_runner.dart:283:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
           
           
[  +66 ms] ensureAnalyticsSent: 66ms
[        ] Running 1 shutdown hook
[   +1 ms] Shutdown hooks complete
[        ] exiting with code 1
```

**flutter doctor -v and flutter run -v did not change after any attempted solution**

### Logs

<details open>
<summary>Logs</summary>

```console

➜  ~ git:(main) ✗ sudo gem install cocoapods
Password:
Successfully installed cocoapods-1.15.2
Parsing documentation for cocoapods-1.15.2
Done installing documentation for cocoapods after 1 seconds
1 gem installed

➜  ~ git:(main) ✗ sudo gem uninstall ffi && sudo gem install ffi -- --enable-libffi-alloc
You have requested to uninstall the gem:
	ffi-1.17.0-x86_64-darwin

ethon-0.16.0 depends on ffi (>= 1.15.0)
If you remove this gem, these dependencies will not be met.
Continue with Uninstall? [yN]  y
Successfully uninstalled ffi-1.17.0-x86_64-darwin
Fetching ffi-1.17.0-x86_64-darwin.gem
Successfully installed ffi-1.17.0-x86_64-darwin
Parsing documentation for ffi-1.17.0-x86_64-darwin
Installing ri documentation for ffi-1.17.0-x86_64-darwin
Done installing documentation for ffi after 0 seconds
1 gem installed

➜ ~ git:(main) ✗ pod --version
1.15.2


➜ ~ git:(main) ✗ sudo gem uninstall cocoapods
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-core
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-try
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-trunk
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-search
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-plugins
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-downloader
➜ ~ git:(main) ✗ sudo gem uninstall cocoapods-deintegrate
Remove executables:
	pod, sandbox-pod

in addition to the gem? [Yn] y
Removing pod
Removing sandbox-pod
Successfully uninstalled cocoapods-1.15.2
Successfully uninstalled cocoapods-core-1.15.2
Successfully uninstalled cocoapods-try-1.2.0
Successfully uninstalled cocoapods-trunk-1.6.0
Successfully uninstalled cocoapods-search-1.0.1
Successfully uninstalled cocoapods-plugins-1.0.0
Successfully uninstalled cocoapods-downloader-2.1
Successfully uninstalled cocoapods-deintegrate-1.0.5

➜ ~ git:(main) ✗ brew install cocoapods
==> Downloading https://formulae.brew.sh/api/formula.jws.json
######################################################################### 100.0%
==> Downloading https://formulae.brew.sh/api/cask.jws.json
######################################################################### 100.0%
Warning: Treating cocoapods as a formula. For the cask, use homebrew/cask/cocoapods or specify the `--cask` flag.
==> Downloading https://ghcr.io/v2/homebrew/core/cocoapods/manifests/1.15.2
Already downloaded: /Users/$USER/Library/Caches/Homebrew/downloads/9d1c37f12921510950311c1778e42d8c156097eec05ac8fb3d075f43f306b475--cocoapods-1.15.2.bottle_manifest.json
==> Fetching dependencies for cocoapods: ruby
==> Downloading https://ghcr.io/v2/homebrew/core/ruby/manifests/3.3.2
Already downloaded: /Users/$USER/Library/Caches/Homebrew/downloads/576482068f581a9a936b82b6c7fd387edc923273bd86c6e58f30ada4bfc1545b--ruby-3.3.2.bottle_manifest.json
==> Fetching ruby
==> Downloading https://ghcr.io/v2/homebrew/core/ruby/blobs/sha256:5ae0365228c91
Already downloaded: /Users/$USER/Library/Caches/Homebrew/downloads/ec3a4f164b16b691a93d576695e3d8c28a1ea8ae0352f376c448ad5fed4d1d9d--ruby--3.3.2.sonoma.bottle.tar.gz
==> Fetching cocoapods
==> Downloading https://ghcr.io/v2/homebrew/core/cocoapods/blobs/sha256:981c5baa
Already downloaded: /Users/$USER/Library/Caches/Homebrew/downloads/c56a47a011dbfdefe1cf46da8ac5175e8fb1e2047063212cddd61cb16657b9eb--cocoapods--1.15.2.sonoma.bottle.tar.gz
==> Installing dependencies for cocoapods: ruby
==> Installing cocoapods dependency: ruby
==> Downloading https://ghcr.io/v2/homebrew/core/ruby/manifests/3.3.2
Already downloaded: /Users/$USER/Library/Caches/Homebrew/downloads/576482068f581a9a936b82b6c7fd387edc923273bd86c6e58f30ada4bfc1545b--ruby-3.3.2.bottle_manifest.json
==> Pouring ruby--3.3.2.sonoma.bottle.tar.gz
:beer: /usr/local/Cellar/ruby/3.3.2: 19,792 files, 46.5MB
==> Installing cocoapods
==> Pouring cocoapods--1.15.2.sonoma.bottle.tar.gz
:beer: /usr/local/Cellar/cocoapods/1.15.2: 13,736 files, 28.8MB
==> Running `brew cleanup cocoapods`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

➜ ~ git:(main) ✗ pod --version
1.15.2

➜ ~ git:(main) ✗ xcode-select --install
xcode-select: note: Command line tools are already installed. Use "Software Update" in System Settings or the softwareupdate command line interface to install updates

```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

[!] Flutter (Channel stable, 3.7.8, on macOS 14.5 23F79 darwin-arm64 (Rosetta),
  locale en-US)
  • Flutter version 3.7.8 on channel stable at /Users/$USER/Desktop/flutter
  ! The flutter binary is not on your path. Consider adding
   /Users/$USER/Desktop/flutter/bin to your path.
  ! The dart binary is not on your path. Consider adding
   /Users/$USER/Desktop/flutter/bin to your path.
  • Upstream repository https://github.com/flutter/flutter.git
  • Framework revision 90c64ed42b (1 year, 2 months ago), 2023-03-21 11:27:08
   -0500
  • Engine revision 9aa7816315
  • Dart version 2.19.5
  • DevTools version 2.20.1
  • If those were intentional, you can disregard the above warnings; however
   it is recommended to use "git" directly to perform update checks and
   upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
  • Android SDK at /Users/$USER/Library/Android/sdk
  • Platform android-33, build-tools 33.0.1
  • Java binary at: /Applications/Android
   Studio.app/Contents/jbr/Contents/Home/bin/java
  • Java version OpenJDK Runtime Environment (build
   11.0.15+0-b2043.56-8887301)
  • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 15.4)
  • Xcode at /Applications/Xcode.app/Contents/Developer
  • Build 15F31d
  ✗ CocoaPods not installed.
    CocoaPods is used to retrieve the iOS and macOS platform side's plugin
    code that responds to your plugin usage on the Dart side.
    Without CocoaPods, plugins will not work on iOS or macOS.
    For more info, see https://flutter.dev/platform-plugins
   To install see
   https://guides.cocoapods.org/using/getting-started.html#installation for
   instructions.

[✓] Chrome - develop for the web
  • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
  • Android Studio at /Applications/Android Studio.app/Contents
  • Flutter plugin can be installed from:
   :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
  • Dart plugin can be installed from:
   :hammer: https://plugins.jetbrains.com/plugin/6351-dart
  • Java version OpenJDK Runtime Environment (build
   11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.89.1)
  • VS Code at /Applications/Visual Studio Code.app/Contents
  • Flutter extension version 3.90.0

[✓] Connected device (3 available)
  • iPhone 15 (mobile) • 17DF0199-F98F-49E2-9418-12A52671F5CC • ios
   • com.apple.CoreSimulator.SimRuntime.iOS-17-5 (simulator)
  • macOS (desktop)  • macos                • darwin-arm64
   • macOS 14.5 23F79 darwin-arm64 (Rosetta)
  • Chrome (web)    • chrome                • web-javascript
   • Google Chrome 125.0.6422.141

[✓] HTTP Host Availability
  • All required HTTP hosts are available

! Doctor found issues in 2 categories.

```

</details>

**Comment by darshankawar** [comment]
@Stephanie-Limber 
I see that you are on old version of stable. I think upgrading it to latest and re-running your scenario should help.

**Comment by Stephanie-Limber** [comment]
Unfortunately, upgrading to flutter 3.22.1 did not resolve the flutter doctor or flutter run errors

```
[!] Flutter (Channel stable, 3.22.1, on macOS 14.5 23F79 darwin-arm64 (Rosetta),
    locale en-US)
    • Flutter version 3.22.1 on channel stable at /Users/$USER/Desktop/flutter
    ! The flutter binary is not on your path. Consider adding
      /Users/$USER/Desktop/flutter/bin to your path.
    ! The dart binary is not on your path. Consider adding
      /Users/$USER/Desktop/flutter/bin to your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a14f74ff3a (13 days ago), 2024-05-22 11:08:21 -0500
    • Engine revision 55eae6864b
    • Dart version 3.4.1
    • DevTools version 2.34.3
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/$USER/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.90.0

[✓] Connected device (4 available)
    • iPhone 15 (mobile)              • 17DF0199-F98F-49E2-9418-12A52671F5CC •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-5 (simulator)
    • macOS (desktop)                 • macos                                •
      darwin-arm64   • macOS 14.5 23F79 darwin-arm64 (Rosetta)
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                •
      darwin         • macOS 14.5 23F79 darwin-arm64 (Rosetta)
    • Chrome (web)                    • chrome                               •
      web-javascript • Google Chrome 125.0.6422.141

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

**Comment by Stephanie-Limber** [comment]
I was able to resolve the issue.

I fully uninstalled homebrew by applying the terminal line here: [Uninstall Homebrew](https://github.com/homebrew/install#uninstall-homebrew)

I also had to forcefully remove /opt/homebrew using `sudo rm -rf /opt/homebrew` to finish uninstalling everything.

Then I restarted my computer and installed homebrew using the `/bin/bash` line in their documentation: [Homebrew Docs](https://brew.sh/)

I followed the output commands and re-added homebrew to my PATH.  I also had to `brew install direnv`, but that might be unrelated.

Then I did `brew install cocoapods.`

`flutter doctor -v` shows this fixed both the PATH issue for the flutter and dart binaries and the cocoapods issue.  I'm not sure I understand how these actions resolved the binary path issue, but I'm just going with it.

Thank you for trying to help!
